src/components/Secrets/secret-actions.ts

```diff
@@ -1,16 +1,21 @@
+import { useIsOnFeatureFlag } from '~/feature-flags/hooks';
 import { SecretModel } from '../../models';
 import { Action } from '../../shared/components/action-menu/types';
 import { SecretKind } from '../../types';
 import { useAccessReviewForModel } from '../../utils/rbac';
 import { useModalLauncher } from '../modal/ModalProvider';
-import { secretDeleteModal } from './secret-modal';
+import { secretDeleteModal, secretDeleteModalForBuildServiceAccount } from './secret-modal';
 
 export const useSecretActions = (secret: SecretKind): Action[] => {
   const showModal = useModalLauncher();
   const [canDelete] = useAccessReviewForModel(SecretModel, 'delete');
+  const isNew = useIsOnFeatureFlag('buildServiceAccount');
   return [
     {
-      cta: () => showModal(secretDeleteModal(secret)),
+      cta: () =>
+        showModal(
+          !isNew ? secretDeleteModal(secret) : secretDeleteModalForBuildServiceAccount(secret),
```

Comment:
nit: It's fine for this PR, But a better variable name would be self explanatory